File: applications/luci-statistics/luasrc/statistics/rrdtool/definitions/conntrack.lua
--[[

Luci statistics - conntrack plugin diagram definition
(c) 2011 Jo-Philipp Wich <xm@subsignal.org>

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

        http://www.apache.org/licenses/LICENSE-2.0

$Id$

]]--

module("luci.statistics.rrdtool.definitions.conntrack",package.seeall)

function rrdargs( graph, plugin, plugin_instance, dtype )
	return {
		title = "%H: Conntrack entries",
		vlabel = "Count",
		number_format = "%5.0lf",
		data = {
			sources = {
				conntrack = { "entropy" }
			},
			options = {
				conntrack = { 
					color = "0000ff",
					title = "Tracked connections"
				}
			}
		}
	}
end

File: applications/luci-app-statistics/htdocs/luci-static/resources/statistics/rrdtool/definitions/conntrack.js
/* Licensed to the public under the Apache License 2.0. */

'use strict';
'require baseclass';

return baseclass.extend({
	title: _('Conntrack'),

	rrdargs: function(graph, host, plugin, plugin_instance, dtype) {
		var entries = {
			title: "%H: Conntrack entries",
			vlabel: "Count",
			number_format: "%5.0lf",
			data: {
				/* collectd 5.5+: specify "" to exclude "max" instance */
				instances: {
					conntrack: [ "" ]
				},
				sources: {
					conntrack: [ "value" ]
				},
				options: {
					conntrack: {
						color: "0000ff",
						title: "Tracked connections"
					}
				}
			}
		};

		var percent = {
			title: "%H: Conntrack usage",
			vlabel: "Percent",
			number_format: "%5.1lf%%",
			y_min: "0",
			alt_autoscale_max: true,
			data: {
				instances: {
					percent: [ "used" ]
				},
				options: {
					percent_used: {
						color: "00ff00",
						title: "Used"
					}
				}
			}
		};

		return [ entries, percent ];
	}
});
